File: 2023_H2/Utmutato/3_Reklam/reklam.cpp
// reklam.cpp

#include <locale>
#include <iostream>
#include <vector>
#include <string>
#include <fstream>

using namespace std;

typedef struct {
	unsigned short nap;
	string varos;
	unsigned short darab;
} rendeles;

vector <rendeles> rendelesek;

void fel1() {
	cout << "1. feladat:" << endl;
	ifstream rendfile("rendel.txt");
	while (not rendfile.eof()) {
		rendeles egyrend;
		rendfile >> egyrend.nap >> egyrend.varos >> egyrend.darab >> ws;
		rendelesek.push_back(egyrend);
	}
	rendfile.close();
}

void fel2() {
	cout << "2. feladat:" << endl;
	cout << "A rendelések száma: " << rendelesek.size() << endl;
}

void fel3() {
	cout << "3. feladat:" << endl;
	unsigned short nap;
	cout << "Kérem adjon meg egy napot: ";
	cin >> nap;
	unsigned short rendszam = 0;
	unsigned short ri = 0;
	while (ri < rendelesek.size() and rendelesek[ri].nap <= nap) {
		if (rendelesek[ri].nap == nap)
			++rendszam;
		++ri;
	}
	cout << "A rendelések száma az adott napon: " << rendszam << endl;
}

void fel4() {
	cout << "4. feladat:" << endl;
	unsigned short nap = 1, nemvolt = 0;

	for (unsigned short ri = 0; ri < rendelesek.size(); ++ri) {
		if (rendelesek[ri].varos == "NR" and rendelesek[ri].nap != nap) {
			nemvolt += rendelesek[ri].nap - nap - 1;
			nap = rendelesek[ri].nap;
		}
	}
	nemvolt += 30 - nap;

	if (nemvolt > 0)
		cout << nemvolt << " nap nem volt a reklámban nem érintett városból rendelés" << endl;
	else 
		cout << "Minden nap volt rendelés a reklámban nem érintett városból" << endl;
}

void fel5() {
	cout << "5. feladat:" << endl;
	unsigned short maxrend = rendelesek[0].darab, maxnap = rendelesek[0].nap;

	for (unsigned short ri = 1; ri < rendelesek.size(); ++ri)
		if (rendelesek[ri].darab > maxrend) {
			maxrend = rendelesek[ri].darab;
			maxnap = rendelesek[ri].nap;
		}
	cout << "A legnagyobb darabszám: " << maxrend << ", a rendelés napja: " << maxnap << endl;
}

unsigned short osszes(string varos, unsigned short nap) {
	unsigned short ossz = 0;
	for (unsigned short ri = 0; ri < rendelesek.size(); ++ri)
		if (rendelesek[ri].varos == varos and rendelesek[ri].nap == nap)
			ossz += rendelesek[ri].darab;
	return ossz;
}

void fel7() {
	cout << "7. feladat:" << endl;
	unsigned short plossz = osszes("PL", 21);
	unsigned short tvossz = osszes("TV", 21);
	unsigned short nrossz = osszes("NR", 21);

	cout << "A rendelt termékek darabszáma a 21. napon PL: " << plossz << " TV: " << tvossz << " NR: " << nrossz << endl;
}

void fel8() {
	cout << "8. feladat:" << endl;
	string varosok[3] = { "PL","TV","NR" };
	unsigned short ossztiz[3][3] = {0,0,0, 0,0,0, 0,0,0};
	for (unsigned short ri = 0; ri < rendelesek.size(); ++ri) {
		unsigned short naptiz = (rendelesek[ri].nap-1)/10;
		unsigned short vartiz = 0;
		while (varosok[vartiz] != rendelesek[ri].varos)
			++vartiz;
		++ossztiz[vartiz][naptiz];
	}
	cout << "Napok\t1..10\t11..20\t21..30" << endl;
	for (unsigned short vartiz = 0; vartiz<3; ++vartiz) {
		cout << varosok[vartiz];
		for(unsigned short naptiz = 0; naptiz<3; ++naptiz) {
			cout << '\t' << ossztiz[vartiz][naptiz];
		}
		cout << endl;
	}
	ofstream kampfile("kampany.txt");
	kampfile << "Napok\t1..10\t11..20\t21..30" << endl;
	for (unsigned short vartiz = 0; vartiz < 3; ++vartiz) {
		kampfile << varosok[vartiz];
		for (unsigned short naptiz = 0; naptiz < 3; ++naptiz) {
			kampfile << '\t' << ossztiz[vartiz][naptiz];
		}
		kampfile << endl;
	}
	kampfile.close();
}

int main() {
	setlocale(LC_ALL,"hu_HU");
	fel1();
	fel2();
	fel3();
	fel4();
	fel5();
	fel7();
	fel8();

	return 0;
}

File: 2023_H2/Útmutató/reklam.py
def fel1():
    print("1. feladat:")
    with open("rendel.txt", 'r') as rendfile:
        rendelesek = []
        for line in rendfile:
            nap, varos, darab = line.split()
            rendelesek.append({'nap': int(nap), 'varos': varos, 'darab': int(darab)})
    return rendelesek

def fel2(rendelesek):
    print("2. feladat:")
    print("A rendelések száma:", len(rendelesek))

def fel3(rendelesek):
    print("3. feladat:")
    nap = int(input("Kérem adjon meg egy napot: "))
    rendszam = sum(1 for rend in rendelesek if rend['nap'] == nap)
    print("A rendelések száma az adott napon:", rendszam)

def fel4(rendelesek):
    print("4. feladat:")
    nap = 1
    nemvolt = 0
    for rend in rendelesek:
        if rend['varos'] == "NR" and rend['nap'] != nap:
            nemvolt += rend['nap'] - nap - 1
            nap = rend['nap']
    nemvolt += 30 - nap
    if nemvolt > 0:
        print(nemvolt, "nap nem volt a reklámban nem érintett városból rendelés")
    else:
        print("Minden nap volt rendelés a reklámban nem érintett városból")

def fel5(rendelesek):
    print("5. feladat:")
    maxrend = max(rendelesek, key=lambda rend: rend['darab'])
    print("A legnagyobb darabszám:", maxrend['darab'], ", a rendelés napja:", maxrend['nap'])

def osszes(rendelesek, varos, nap):
    ossz = sum(rend['darab'] for rend in rendelesek if rend['varos'] == varos and rend['nap'] == nap)
    return ossz

def fel7(rendelesek):
    print("7. feladat:")
    plossz = osszes(rendelesek, "PL", 21)
    tvossz = osszes(rendelesek, "TV", 21)
    nrossz = osszes(rendelesek, "NR", 21)
    print("A rendelt termékek darabszáma a 21. napon PL:", plossz, "TV:", tvossz, "NR:", nrossz)

def fel8(rendelesek):
    print("8. feladat:")
    varosok = ["PL", "TV", "NR"]
    ossztiz = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
    for rend in rendelesek:
        naptiz = (rend['nap'] - 1) // 10
        vartiz = varosok.index(rend['varos'])
        ossztiz[vartiz][naptiz] += 1

    with open("kampany.txt", 'w') as kampfile:
        kampfile.write("Napok\t1..10\t11..20\t21..30\n")
        for vartiz in range(len(varosok)):
            kampfile.write(varosok[vartiz])
            for naptiz in range(3):
                kampfile.write('\t' + str(ossztiz[vartiz][naptiz]))
            kampfile.write('\n')

def main():
    rendelesek = fel1()
    fel2(rendelesek)
    fel3(rendelesek)
    fel4(rendelesek)
    fel5(rendelesek)
    fel7(rendelesek)
    fel8(rendelesek)

if __name__ == "__main__":
    main()
